src/v/redpanda/admin_server.cc

```diff
@@ -291,12 +298,31 @@ ss::future<> admin_server::configure_listeners() {
                         logger, "API TLS", updated, eptr);
                   });
             }
+
+            if (!localhost && !tls_it->config.get_require_client_auth()) {
```

Comment:
do we want to distinguish between "unauthenticated listener" (from the commit message) and "unsecured listener"? For example, `config.get_require_client_auth` is presumably referring to mTLS but we still rely on SCRAM for authentication. Similarly, I would presume (but don't know for sure) that we can have TLS without mutual auth (e.g. letsencrypt setups or other public certs).

Reply:
The distinction I'm making is that "insecure" is the combination of being unauthenticated and listening on an external address.  Then I'm using "authentication" to refer to one or both of SCRAM and mTLS (where the client presents a valid certificate, it has authenticated itself, we just happen to ignore the identity at present).

Reply:
I guess what I meant was: won't this result in a WARN if I'm using anything other than mTLS (e.g. basic auth with or without TLS) even though it fits the requirement of authentication?

Reply:
Argh, you're right, this was meant to first check for an insecure listener, and then only emit the warning below if admin_api_require_auth was also disabled, and I apparently blanked on actually writing the second part.  Fixing.